networks/3a_validate_array_vs_rnaseq_modules.py

```diff
@@ -0,0 +1,217 @@
+
+# %%time
+# Plot heatmap
+"""o1 = sns.clustermap(pao1_array_corr_original.abs(), cmap="viridis", figsize=(20, 20))
+o1.fig.suptitle("Correlation of raw PAO1 genes (array compendium)", y=1.05)"""
+
+# Transpose compendia to be gene x sample
+# Here we're interested in how genes cluster
+pao1_array_compendium_T = pao1_array_compendium.T
+pao1_rnaseq_compendium_T = pao1_rnaseq_compendium.T
+
+# log transform data
+pao1_array_compendium_log10 = np.log10(1 + pao1_array_compendium_T)
+pao1_rnaseq_compendium_log10 = np.log10(1 + pao1_rnaseq_compendium_T)
+
+# Apply SVD
+array_U, array_s, array_Vh = np.linalg.svd(
+    pao1_array_compendium_log10, full_matrices=False
+)
+rnaseq_U, rnaseq_s, rnaseq_Vh = np.linalg.svd(
+    pao1_rnaseq_compendium_log10, full_matrices=False
+)
+
+# Convert ndarray to df to use corr()
+array_U_df = pd.DataFrame(data=array_U, index=pao1_array_compendium_T.index)
+rnaseq_U_df = pd.DataFrame(data=rnaseq_U, index=pao1_rnaseq_compendium_T.index)
+
+# Correlation of U
+# Since `corr()` computes pairwise correlation of columns we need to invert U
+pao1_array_corr_log_spell = array_U_df.iloc[:, :num_singular_values_log].T.corr()
+pao1_rnaseq_corr_log_spell = rnaseq_U_df.iloc[:, :num_singular_values_log].T.corr()
+
+# Note: Here we plot the heatmaps to verify that the correlation of log + SPELL transformed data looks as expected (i.e. there is not a single large cluster)
+
+# Plot heatmap
+"""h1a = sns.clustermap(pao1_array_corr_log_spell.abs(), cmap="viridis", figsize=(20, 20))
+h1a.fig.suptitle(
+    f"log transform + SPELL corrected using {num_singular_values_log} vectors (PAO1 array)",
+    y=1.05,
+)"""
+
+# Plot heatmap
+"""h1b = sns.clustermap(pao1_rnaseq_corr_log_spell.abs(), cmap="viridis", figsize=(20, 20))
+h1b.fig.suptitle(
+    f"log transform + SPELL corrected using {num_singular_values_log} vectors (PAO1 rnaseq)",
+    y=1.05,
+)"""
+
+# ## Clustering and get module membership
+
+# Clustering using DBSCAN
+if cluster_method == "dbscan":
+    pao1_array_clustering = DBSCAN(eps=density_threshold).fit(pao1_array_corr_log_spell)
+    pao1_rnaseq_clustering = DBSCAN(eps=density_threshold).fit(
+        pao1_rnaseq_corr_log_spell
+    )
+
+# Clustering using hierarchal clustering
+if cluster_method == "hierarchal":
+    pao1_array_clustering = AgglomerativeClustering(
+        n_clusters=None, distance_threshold=hier_threshold, linkage=link_dist
+    ).fit(ppao1_array_corr_log_spell)
+    pao1_rnaseq_clustering = AgglomerativeClustering(
+        n_clusters=None, distance_threshold=hier_threshold, linkage=link_dist
+    ).fit(pao1_rnaseq_corr_log_spell)
+
+# Clustering using affinity propogation
+if cluster_method == "affinity":
+    pao1_array_clustering = AffinityPropagation(random_state=0).fit(
+        pao1_array_corr_log_spell
+    )
+    pao1_rnaseq_clustering = AffinityPropagation(
+        random_state=1, damping=affinity_damping
+    ).fit(pao1_rnaseq_corr_log_spell)
```

Comment:
Here `pao1_array_corr_log_spell`, the input to all algorithms, is a squared matrix of correlations, right? I see that `AffinityPropagation` supports features or affinities as input, but you have to provide the `affinity='precomputed'` parameter. Not sure if I'm missing something.

Reply:
Another question: why are you using a different set of parameters here for RNA-seq and array data?

Reply:
Good point. I'll need to think a bit more about the parameter settings I'm using. The reason for the difference is because the model didn't converge unless i increased the damping factor here.